components/map/EventMarker.js

```diff
@@ -0,0 +1,42 @@
+import L from "leaflet";
+import { Marker, Popup } from "react-leaflet";
+import { ReactMarkdown } from "react-markdown/lib/react-markdown";
+import Link from "@components/Link";
+
+export default function EventMarker({event}) {
+  // Custom component for rendering links within ReactMarkdown
+  const LinkRenderer = ({ href, children }) => (
+    <Link href={href}>
+      {children}
+    </Link>
+  );
+
+  return (
+    <Marker
+      icon={L.icon({
+        className: "rounded-full",
+        iconUrl: `/placard.png`,
+        popupAnchor: [0, -10],
+        iconSize: [40, 40],
+        iconAnchor: [20, 20],
+      })}
+      position={[event.geometry.coordinates[1], event.geometry.coordinates[0]]}
+    >
+      <Popup>
+        <div className="flex flex-col gap-[5px]">
+          <h1 className="font-[600]">
+            <Link
+              href={`${event.properties.url}`}
+            >
+              {event.properties.name}
+            </Link>
+          </h1>
+          <span>{event.properties.location.city}</span>
```

Comment:
This should be changed to use city, state, & country.
Not everyone is using the location fields in the same way (or in the way that you or I think is correct).
Something like this should work:
```js
[
  event.properties.location.city,
  event.properties.location.state,
  event.properties.location.country
].filter((x) => x).join(", ");
```

It would also be nice to show the dates somewhere on the popup.

Reply:
![image](https://github.com/EddieHubCommunity/LinkFree/assets/68022411/f47acecb-ebd9-4c89-8052-b9f27755e944)
I have formatted it like this.